2022/Q2/artifacts/pos-model/PoS-model.md

```diff
@@ -191,7 +180,7 @@ tx_withdraw_unbonds_delegator(delegator_address)
 //the only possible values for offset_length are pipeline_length and ubonding_length
 func bond(validator_address, delegator_address, amount, offset_length)
 {
-  if is_validator(validator_address) then
+  if is_validator(validator_address, cur_epoch+offset_length) then
```

Comment:
How will you know that the validator is a validator at` cur_epoch + offset_length` if we only set it to candidate status in a previous `prev_epoch` (`prev_epoch < cur_epoch`) and make it a candidate at `prev_epoch + pipeline_legnth` ?

Reply:
Not sure I totally understood but this may help: function `is_validator` uses function `read_epoched_field`

Reply:
Actually, your comment made my reflect about the following.
When we run `tx_deactivate`, we don't look up all the possible `bond` operations that are still ongoing (i.e., no unbond done yet).
So, all those bonds are somehow dangling. The model works because, `tx_undelegate` and `tx_unbond` don't check the status, so the delegator can use them to get the tokens back.
However, what does it mean to have an "ongoing" bonding on a validator that was deactivated long ago? And, when the delegator finally executes `tx_undelegate`, shall it still wait for the unbonding period?

Reply:
Good point. I have opened an issue about this: https://github.com/informalsystems/partnership-heliax/issues/10